### ISSUE TITLE
Octave's subsref seems to return weird sized empty matrices

### ISSUE THREAD

**Issue by genuinelucifer** [cbm755/octsympy#709]
I was trying to fix the subsref behaviour for logical indices and ran into the following problem.
.
If we return empty matrix on this `a([false false])` where `a` is a `sym` then the size of that matrix will be `0x0` while octave returns (for `a` as `double`) a matrix of size `1x0`!! If the matrix is empty then does `1` in `1x0` make sense? Moreover, I can't replicate such matrices without hardcopying them, i.e., `sym(a[false false])`....
After the changes to fix logical indices, the results are:
```
>> a = [1 2 3 5; 4 5 6 9; 7 5 3 2];
>> b = sym(a);
>> c = [false false];
>> a(c)
ans = [](1x0)
>> b(c)
ans = [](0x0)
```

**Comment by genuinelucifer** [comment]
Octave actually returns a "0x0" empty matrix if a scalar false is used in subsref.  
"1x0" empty matrix is returned if a row vector of all false is use.   
"0x1" empty matrix is returned for all other cases where all the values are false.   
Should we also follow this? Also how to produce an empty sym of such dimensions?

**Comment by cbm755** [comment]
`sym('a', [0 1])` and `sym('a', [1 0])` and `sym('a', [0 0])`.

Actually, maybe `sym(zeros(0,1))` is nicer?

**Comment by cbm755** [comment]
> Should we also follow this?

probably we should.  There are some failing tests: are those related?

**Comment by genuinelucifer** [comment]
> There are some failing tests: are those related?

Yes three tests are failing because the result of same operation on double is not equal to the sym counterpart. I'll fix those and add the tests.

**Comment by cbm755** [comment]
Is this closed?  I don't think its a problem in core Octave but we still have errors here right?

**Comment by genuinelucifer** [comment]
I thought it's weird (and probably erroneous) behavior of octave. So I opened this. Since we decided to follow the same here so I closed this. I'll update #710 by tomorrow to follow this.

**Comment by cbm755** [comment]
Fixed by 479522393a909ba1f199eddb77b576b960b4ccd1, thanks!
